diff --git a/tools/gocc/testdata/test24.go b/tools/gocc/testdata/test24.go
index 5355977..7d708e0 100644
--- a/tools/gocc/testdata/test24.go
+++ b/tools/gocc/testdata/test24.go
@@ -8,14 +8,18 @@
 //limitations under the License.
 package main
 
-import "sync"
+import (
+	"sync"
+	rtm "github.com/uber-research/GOCC/tools/gocc/rtmlib"
+)
 
 var m [2]sync.Mutex
 
 func HoH() {
+	optiLock1 := rtm.OptiLock{}
 	m[0].Lock()
-	m[1].Lock()
-	m[0].Unlock()
+	optiLock1.Lock(&m[1])
+	optiLock1.Unlock(&m[0])
 	m[1].Unlock()
 }
 
