diff --git a/tools/gocc/testdata/test17.go b/tools/gocc/testdata/test17.go
index 7d250ff..a335f20 100644
--- a/tools/gocc/testdata/test17.go
+++ b/tools/gocc/testdata/test17.go
@@ -8,12 +8,16 @@
 //limitations under the License.
 package main
 
-import "sync"
+import (
+	"sync"
+	rtm "github.com/uber-research/GOCC/tools/gocc/rtmlib"
+)
 
 func foo(m, n *sync.Mutex) {
+	optiLock1 := rtm.OptiLock{}
 	m.Lock()
-	n.Lock()
-	n.Unlock()
+	optiLock1.Lock(n)
+	optiLock1.Unlock(n)
 	m.Unlock()
 }
 
