diff --git a/tools/gocc/testdata/test22.go b/tools/gocc/testdata/test22.go
index 1ee1703..7862865 100644
--- a/tools/gocc/testdata/test22.go
+++ b/tools/gocc/testdata/test22.go
@@ -8,18 +8,23 @@
 //limitations under the License.
 package main
 
-import "sync"
+import (
+	"sync"
+	rtm "github.com/uber-research/GOCC/tools/gocc/rtmlib"
+)
 
 func bar() {
+	optiLock1 := rtm.OptiLock{}
 	var n sync.Mutex
-	n.Lock()
-	n.Unlock()
+	optiLock1.Lock(&n)
+	optiLock1.Unlock(&n)
 }
 func foo() {
+	optiLock1 := rtm.OptiLock{}
 	var m sync.Mutex
-	m.Lock()
+	optiLock1.Lock(&m)
 	bar()
-	m.Unlock()
+	optiLock1.Unlock(&m)
 }
 
 // test different types of Locks
