diff --git a/tools/gocc/testdata/test11.go b/tools/gocc/testdata/test11.go
index 37e628c..3afd542 100644
--- a/tools/gocc/testdata/test11.go
+++ b/tools/gocc/testdata/test11.go
@@ -9,6 +9,7 @@
 package main
 
 import (
+	rtm "github.com/uber-research/GOCC/tools/gocc/rtmlib"
 	"sync"
 )
 
@@ -16,11 +17,13 @@ var global int
 
 // m should not be replaced since it contains nested lock
 func main() {
+	optiLock2 := rtm.OptiLock{}
+	optiLock1 := rtm.OptiLock{}
 	var m sync.Mutex
 	var n sync.Mutex
-	m.Lock()
-	n.Lock()
+	optiLock2.Lock(&m)
+	optiLock1.Lock(&n)
 	global++
-	n.Unlock()
-	m.Unlock()
+	optiLock1.Unlock(&n)
+	optiLock2.Unlock(&m)
 }
