diff --git a/tools/gocc/testdata/test31.go b/tools/gocc/testdata/test31.go
index 59e86ae..1ef99b8 100644
--- a/tools/gocc/testdata/test31.go
+++ b/tools/gocc/testdata/test31.go
@@ -1,13 +1,15 @@
 package main
 
 import (
+	rtm "github.com/uber-research/GOCC/tools/gocc/rtmlib"
 	"sync"
 )
 
 func foo(m *sync.Mutex, u func()) {
-	m.Lock()
+	optiLock1 := rtm.OptiLock{}
+	optiLock1.Lock(m)
 	u()
-	m.Unlock()
+	optiLock1.Unlock(m)
 }
 
 func main() {
